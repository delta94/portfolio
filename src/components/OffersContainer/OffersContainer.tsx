import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { CustomTheme } from '../../theme';

export interface OffersContainerProps {
  className?: string;
  title?: string;
  offerings: string[];
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    // textAlign: 'center',
    border: `1px solid ${theme.colors.secondaryColor}`,
    borderRadius: 10,
    // justifyContent: 'center',
    width: '100%',
    paddingTop: theme.spacing(1),
    marginTop: theme.spacing(1),
    paddingLeft: theme.spacing(5),
    paddingRight: theme.spacing(5),
  },

  header: {
    color: theme.colors.white,
    borderBottomColor: theme.colors.thirdColor,
    borderBottomWidth: 3,
    borderBottomStyle: 'dashed',
    paddingBottom: theme.spacing(2),
    textAlign: 'center',
    paddingTop: theme.spacing(2),
  },

  ulContainer: {
    listStyleType: 'circle',
    color: theme.colors.white,
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginTop: theme.spacing(2.5),
  },
}));

const OffersContainer: React.FC<OffersContainerProps> = ({ className, title, offerings }) => {
  const classes = useStyles();

  return (
    <div className={clsx([classes.mainContainer, className])}>
      <h3 className={classes.header}>{title}</h3>
      <ul className={classes.ulContainer}>
        {offerings.map((offering: string) => {
          return <li key={offering}>{offering}</li>;
        })}
      </ul>
    </div>
  );
};

export { OffersContainer };
