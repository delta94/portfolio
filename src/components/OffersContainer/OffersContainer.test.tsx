import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { OffersContainer } from './OffersContainer';

const mockData = {
  title: 'mock title',
  offerings: ['mock description'],
};

test('OffersContainer renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <OffersContainer {...mockData} />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
