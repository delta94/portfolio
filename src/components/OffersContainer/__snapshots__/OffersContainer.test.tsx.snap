// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-1"
  >
    <h3
      class="header-0-2-2"
    >
      mock title
    </h3>
    <ul
      class="ulContainer-0-2-3"
    >
      <li>
        mock description
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
